packages/backend/src/routes/courses.ts

```diff
@@ -39,7 +39,35 @@ coursesRouter.get("/:id", async (ctx, next) => {
             );
         });
 
-        ctx.body = rows;
+        const assignments = await new Promise((resolve, reject) => {
+            db.all(
+                `SELECT * FROM assignments WHERE course_id = ${id} ORDER BY deadline ASC`,
+                [],
+                (err, rows) => {
+                    if (err) {
+                        reject(err);
+                    } else {
+                        resolve(rows);
+                    }
+                },
+            );
+        });
+
+        const lectures = await new Promise((resolve, reject) => {
+            db.all(
+                `SELECT * FROM lectures WHERE course_id = ${id} ORDER BY date ASC`,
+                [],
+                (err, rows) => {
+                    if (err) {
+                        reject(err);
+                    } else {
+                        resolve(rows);
+                    }
+                },
+            );
+        });
+
+        ctx.body = { course, assignments, lectures };
```

Comment:
@Nurou Route is for the course details page so I think it makes sense to send everything we need there together. We are not going to load assignments and lectures specific to one course anywhere else either, so having different endpoints might be overkill.

But having different routes is valid too, though I think in that case I'd choose /courses/:id/assignments and /courses/:id/lectures for the URL. We could actually load all three in parallel since we do not need to wait for /courses/:id to get the course id.

What do you think?

Reply:
Yes, the routing suggestion you proposed makes more sense. I think it's better we do it that way since it be useful for comparison with graphql client calls where we could specify all the course data we want (inc. lectures and assignments) in one go. 
```gql
course(:id) {
	// course info
	assignments {
		// assignment info
	}
	lectures {
		// lectures info
	}
}

```